src/csync/csync.cpp

```diff
@@ -525,7 +429,7 @@ int csync_destroy(CSYNC *ctx) {
   SAFE_FREE(ctx->local.uri);
```

Comment:
can we put all of this in the destructor of CSYNC where it belongs?

Reply:
I tried to move this stuff into constructor/destructor but the whole csync_init, csync_commit and csync_destroy stuff wouldn't match very well to that concept and I thought I'd just leave things like they were and just do the `new` to get my std::maps constructed.

This could maybe help with that clang crash though, could be worth a try.